src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityRequest.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal class ManagedIdentityRequest
+    {
+        public UriBuilder UriBuilder { get; }
```

Comment:
nit: Have you considered to expose `endpoint` and a `Dictionary<string, string>() getParams` ?